addons/project/models/project.py

```diff
@@ -730,7 +730,7 @@ def _read_group_stage_ids(self, stages, domain, order):
         index=True, tracking=True)
     partner_id = fields.Many2one('res.partner',
         string='Customer',
-        compute='_compute_partner_id', recursive=True, store=True, readonly=False, tracking=True,
```

Comment:
This actually reverts the fix introduced at: https://github.com/odoo-dev/odoo/commit/34d6f87d5497182d8ce395a8c90277f5ddde8a95#diff-dcdc6e8a2e65aff0582c53d72c3890c1b56725a8b9fcf7abbb0378097b5999afR602

Why did you have to make this change ?

Reply:
because of this code (https://github.com/odoo/enterprise/pull/19812/files#diff-505826809b4301767ac93163dcfc8e5a148d197f2100cf6f1bb7e0aface0681bR39) I have to remove recursive from field.

Because we need a specific case that, when we choose a customer then his sales orders should automatically filter for 'sales order item' selection and also when I remove a partner then it should show all the sales order. after that if I select one from the order then it should also filter the customer as per chosen sale order item.